### PR TITLE
dosage is now gp, E(gp) is now dosage

### PR DESCRIPTION
**Pull request by jbloom22**

### CODE REVIEW

File: python/hail/representation/genotype.py [hunk 1]
```diff
@@ -138,12 +138,21 @@ def od(self):
 
         return from_option(self._jrep.od())
 
-    def dosage(self):
+    def gp(self):
```
Comment (tpoterba) [inline python/hail/representation/genotype.py]: add @property here

File: python/hail/representation/genotype.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def __repr__(self):
         fake_ref = 'FakeRef=True' if self._jrep.fakeRef() else ''
         if self._jrep.isDosage():
             return 'Genotype(GT=%s, AD=%s, DP=%s, GQ=%s, PP=%s%s)' %\
```
Comment (tpoterba) [inline python/hail/representation/genotype.py]: Replace "PP" with "GP"

File: src/main/scala/is/hail/variant/Genotype.scala [hunk 3]
```diff
@@ -78,21 +78,23 @@ abstract class Genotype extends Serializable {
     else
```
Comment (tpoterba) [inline src/main/scala/is/hail/variant/Genotype.scala]: (not a comment on this line)
replace isDosage with isLinearScaled or something.

File: src/main/scala/is/hail/variant/Genotype.scala [hunk 4]
```diff
@@ -78,21 +78,23 @@ abstract class Genotype extends Serializable {
...
-  def unboxedDosage: Array[Double] =
+  def unboxedGP: Array[Double] =
     if (unboxedPX == null)
       null
     else if (isDosage)
-      unboxedPX.map(_ * Genotype.dosageNorm)
+      unboxedPX.map(_ * Genotype.gpNorm)
     else
-      Genotype.phredToDosage(unboxedPX)
+      Genotype.plToGP(unboxedPX)
```
Comment (tpoterba) [inline src/main/scala/is/hail/variant/Genotype.scala]: This should be `phredToLinear` or something

File: src/main/scala/is/hail/variant/Genotype.scala [hunk 5]
```diff
@@ -667,13 +675,13 @@ object Genotype {
...
 
-  def phredToDosage(a: Array[Int]): Array[Double] = {
+  def plToGP(a: Array[Int]): Array[Double] = {
     val lkhd = a.map(i => phredToLinear(i))
     val s = lkhd.sum
     lkhd.map(_ / s)
   }
 
-  def phredToBiallelicDosageGenotype(px0: Int, px1: Int, px2: Int): Double = {
+  def phredValsToBiallelicDosage(px0: Int, px1: Int, px2: Int): Double = {
```
Comment (tpoterba) [inline src/main/scala/is/hail/variant/Genotype.scala]: plToDosage?

File: src/main/scala/is/hail/variant/Genotype.scala [hunk 6]
```diff
@@ -863,7 +871,7 @@ object Genotype {
     gt
   }
 
-  def readBiallelicDosageGenotype(isDosage: Boolean, a: ByteIterator): Double = {
+  def readBiallelicDosage(isDosage: Boolean, a: ByteIterator): Double = {
```
Comment (tpoterba) [inline src/main/scala/is/hail/variant/Genotype.scala]: readDosage

File: src/main/scala/is/hail/variant/GenotypeStream.scala [hunk 7]
```diff
@@ -22,12 +22,12 @@ class HardCallGenotypeStreamIterator(nAlleles: Int, isDosage: Boolean, b: ByteIt
   override def next(): Int = Genotype.readHardCallGenotype(nAlleles, isDosage, b)
 }
 
-class BiallelicDosageGenotypeStreamIterator(nAlleles: Int, isDosage: Boolean, b: ByteIterator) extends HailIterator[Double] {
+class BiallelicDosageStreamIterator(nAlleles: Int, isDosage: Boolean, b: ByteIterator) extends HailIterator[Double] {
```
Comment (tpoterba) [inline src/main/scala/is/hail/variant/GenotypeStream.scala]: Just call this "DosageStreamIterator".

File: src/main/scala/is/hail/variant/package.scala [hunk 8]
```diff
@@ -30,13 +30,13 @@ package object variant {
         }
     }
 
-    def biallelicDosageGenotypeIterator: HailIterator[Double] = ig match {
-      case gs: GenotypeStream => gs.gsBiallelicDosageGenotypeIterator
+    def biallelicDosageIterator: HailIterator[Double] = ig match {
```
Comment (tpoterba) [inline src/main/scala/is/hail/variant/package.scala]: kill the biallelic

**Comment by tpoterba** [comment]
Also rebase to breakingbad

File: src/main/scala/is/hail/variant/Genotype.scala (hunk 4 above)
Comment (jbloom22) [inline src/main/scala/is/hail/variant/Genotype.scala]: I disagree.  There is also a normalization here. It talks the PL array to the GP array.

File: src/main/scala/is/hail/variant/Genotype.scala (hunk 5 above)
Comment (jbloom22) [inline src/main/scala/is/hail/variant/Genotype.scala]: agreed

File: src/main/scala/is/hail/variant/Genotype.scala (hunk 6 above)
Comment (jbloom22) [inline src/main/scala/is/hail/variant/Genotype.scala]: good call, I've removed biallelic everywhere.

**Comment by jbloom22** [comment]
moving to PR against breaking bad instead

**Pull request closed without merging**